src/System.Management.Automation/engine/Modules/PSModuleInfo.cs

```diff
@@ -834,6 +837,22 @@ internal void AddToCompatiblePSEditions(string psEdition)
             _compatiblePSEditions.Add(psEdition);
         }
 
+        internal void AddToCompatiblePSEditions(IEnumerable<string> psEditions)
+        {
+            _compatiblePSEditions.AddRange(psEditions);
+        }
+
+        /// <summary>
+        /// Describes whether the module was considered compatible at load time.
+        /// Any module not on the System32 module path should have this as true.
+        /// Modules loaded from the System32 module path will have this as true if they
+        /// have declared edition compatibility with PowerShell Core. Currently, this field
+        /// is true for all non-psd1 module files, when it should not be. Being able to
+        /// load psm1/dll modules from the System32 module path without needing to skip
+        /// the edition check is considered a bug and should be fixed.
```

Comment:
Please open an issue tracking this.

Reply:
Please see https://github.com/PowerShell/PowerShell/issues/7296